### PR TITLE
Quick start

### LINKED ISSUE #2016: Basic documentation for 'got' object
I don't know if I'm looking in the wrong place, but although this seems to be a pretty well-documented project, I am completely unable to find the basic documentation on the got object itself, such as the parameters for `got.post()` and other similar methods (presumably `get`, `put`, `patch`, etc.).

I've read through everything in the documentation folder and while its very useful to know how to manage streams, I'm struggling to find the basics, like whether I need to set a header when posting json or if that's automatic. 

I'm sure it's here somewhere, I just can't find it.

### PR DESCRIPTION
**Pull request by ydarma**
#### Checklist

- [X] I have read the documentation.
- [X] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates in both the README and the types.

Quick start guide. Fix #2016.

### CODE REVIEW

File: documentation/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const url = "https://httpbin.org/anything";
+const data = await got(url).json();
+```
+
+The similar <code>[got.text](1-promise.md#promisetext)</code> method returns plain text.
+
+All `got` methods accepts an options object for passing extra informations, such as headers :
+
+```js
+const url = "https://httpbin.org/anything";
```
Comment (sindresorhus) [inline documentation/quickstart.md:28]: Use single-quotes

File: documentation/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+const options = {
+	headers: {
+		"Custom-Header": "Quick start",
+	},
+	timeout: { send: 3500 },
+};
```
Comment (sindresorhus) [inline documentation/quickstart.md:34]:
```suggestion
};

```

spacing

File: documentation/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+All `got` methods accepts an options object for passing extra informations, such as headers :
+
+```js
+const url = "https://httpbin.org/anything";
+const options = {
+	headers: {
+		"Custom-Header": "Quick start",
+	},
+	timeout: { send: 3500 },
```
Comment (sindresorhus) [inline documentation/quickstart.md:33]:
```suggestion
	timeout: {
		send: 3500
	},
```

File: documentation/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+const options = {
+	json: { documentName: "Quick Start" },
+};
+const outStream = fs.createWriteStream("anything.json");
+got.stream.post(url, options).pipe(outStream);
+```
+
+## Options
+
+Options can be set at client level and reused in subsequent queries :
```
Comment (sindresorhus) [inline documentation/quickstart.md:68]:
```suggestion
Options can be set at client level and reused in subsequent queries:
```

File: documentation/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+	.get("https://httpbin.org/status/404")
+	.catch((e) => console.error(e.code, e.message));
+```
+
+```js
+import got from "got";
+
+got.stream
+	.get("https://httpbin.org/status/404")
+	.once("error", (e) => console.error(e.code, e.message))
```
Comment (sindresorhus) [inline documentation/quickstart.md:111]:
```suggestion
	.once("error", error => {
		console.error(error.code, error.message);
	)
```

File: readme.md [hunk 6]
```diff
@@ -143,6 +143,7 @@ npm install got
 **Warning:** This package is native [ESM](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) and no longer provides a CommonJS export. If your project uses CommonJS, you'll have to [convert to ESM](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) or use the [dynamic `import()`](https://v8.dev/features/dynamic-import) function. Please don't open issues for questions regarding CommonJS / ESM. You can also use [Got v11](https://github.com/sindresorhus/got/tree/v11.8.3) instead which is pretty stable.
 
 ## Take a peek
+**A [quick start](documentation/quickstart.md) guide is available.**
```
Comment (sindresorhus) [inline readme.md:146]:
```suggestion

**A [quick start](documentation/quickstart.md) guide is available.**
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: documentation/quickstart.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
+# Quick start guide
+
+This quick start uses ES2017 syntax.
```
Comment (sindresorhus) [inline documentation/quickstart.md:3]: This is not relevant.

File: documentation/quickstart.md (hunk 5 above)
Comment (ydarma) [reply documentation/quickstart.md]: Done, with a closing `}`

**Comment by sindresorhus** [comment]
All my comments apply in multiple places. I did not comment on all the places, but it's implied.

File: documentation/quickstart.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+import got from 'got';
+import fs from 'fs';
+
+const url = 'https://httpbin.org/anything';
+const options = {
+	json: { documentName: 'Quick Start' },
+};
+
+const outStream = fs.createWriteStream('anything.json');
+got.stream.post(url, options).pipe(outStream);
```
Comment (sindresorhus) [inline documentation/quickstart.md:66]: The recommended Node.js practice is to use the `pipeline` method.

File: documentation/quickstart.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+## Errors
+
+Both Promise and Stream APIs throws error with metadata. They are handled according to the API used.
+
+```js
+import got from 'got';
+
+const data = await got
+	.get('https://httpbin.org/status/404')
+	.catch(error => {
```
Comment (sindresorhus) [inline documentation/quickstart.md:107]: Use async/await with try/catch.

**Comment by ydarma** [comment]
> All my comments apply in multiple places. I did not comment on all the places, but it's implied.

I understood, I tried to apply suggestions everywhere but maybe I missed something...

**Comment by sindresorhus** [comment]
I have pushed some minor improvements: https://github.com/sindresorhus/got/pull/2059/commits/a4fed5a4339b1d43335b30c2c534f5650c3c1e30

**Comment by sindresorhus** [comment]
I think the text still needs improvements. There were many typos and there are probably some left too. Try to go through the text and improve it.

File: documentation/quickstart.md [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+The request body is passed in the options object. The `json` property will automatically set headers accordingly. Custom headers can be added exactly as above.
+
+## Using streams
+
+The [Stream API](3-streams.md) allows to leverage [Node.js Streams](https://nodejs.dev/learn/nodejs-streams) capabilities:
+
+```js
+import got from 'got';
+import fs from 'node:fs';
+import {pipeline} from 'node:stream';
```
Comment (sindresorhus) [inline documentation/quickstart.md:69]: Use the promise version of `pipeline`.

File: documentation/quickstart.md [hunk 11]
```diff
@@ -0,0 +1,233 @@
+# Quick Start Guide
```
Comment (sindresorhus) [inline documentation/quickstart.md:1]: The file should be named `quick-start.md`

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by ydarma** [comment]
> I think the text still needs improvements.

I read the text multiple times and I am afraid I am not able to improve it. I am stuck with my English. If you think about writing style, sorry I won't be able to go further. If you think about more detailed information I can try to go into details. However I think that the quick start should be kept short with pointers to the doc.

Thank for the fixes, I admit that I could have avoid some if I were more careful, but not all of them.

**Comment by sindresorhus** [comment]
You could use something like https://grammarly.com

**Comment by ydarma** [comment]
Thanks, I'll try that ASAP

**Comment by ydarma** [comment]
I applied Grammarly suggestions. The tool showed excellent indicator figures (clarity, engagement, and delivery).

File: documentation/quick-start.md [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+
+The request body is passed in the options object. The `json` property will automatically set headers accordingly. Custom headers can be added exactly as above.
+
+## Using streams
+
+The [Stream API](3-streams.md) allows to leverage [Node.js Streams](https://nodejs.dev/learn/nodejs-streams) capabilities:
+
+```js
+import fs from 'node:fs';
+import {pipeline} from 'node:stream';
```
Comment (sindresorhus) [inline documentation/quick-start.md:68]:
```suggestion
import {pipeline} from 'node:stream/promises';
```

And use the promise version of `pipeline`.

Comment (ydarma) [reply documentation/quick-start.md]: You're right, this is better. I also used promise version of `fs`

File: documentation/quick-start.md [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+The request body is passed in the options object. The `json` property will automatically set headers accordingly. Custom headers can be added exactly as above.
+
+## Using streams
+
+The [Stream API](3-streams.md) allows to leverage [Node.js Streams](https://nodejs.dev/learn/nodejs-streams) capabilities:
+
+```js
+import { open } from "node:fs/promises";
```
Comment (sindresorhus) [inline documentation/quick-start.md:67]:
```suggestion
import {open} from 'node:fs/promises';
```

File: documentation/quick-start.md [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+import { pipeline } from "node:stream/promises";
+import got from "got";
+
+const url = "https://httpbin.org/anything";
```
Comment (sindresorhus) [inline documentation/quick-start.md:71]: Single quotes

File: documentation/quick-start.md [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+
+const options = {
+  json: {
```
Comment (sindresorhus) [inline documentation/quick-start.md:74]: Tab indentation

File: documentation/quick-start.md [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+    documentName: "Quick Start",
+  },
+};
+
+const gotStream = got.stream.post(url, options);
+
+const fd = await open("anything.json", "w");
```
Comment (sindresorhus) [inline documentation/quick-start.md:81]:
```suggestion
const fileDescriptor = await open("anything.json", "w");
```

**Review by sindresorhus — APPROVED** [review]

File: documentation/quick-start.md [hunk 17]
```diff
@@ -0,0 +1,236 @@
...
+
+const options = {
+	json: {
+		documentName: 'Quick Start',
+	},
+};
+
+const gotStream = got.stream.post(url, options);
+
+const fileDescriptor = await open('anything.json', 'w');
```
Comment (szmarczak) [inline documentation/quick-start.md:81]: Does the file descriptor need to be closed after pipeline?

File: documentation/quick-start.md [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+
+Both Promise and Stream APIs throw errors with metadata.
+
+```js
+import got from 'got';
+
+try {
+	const data = await got.get('https://httpbin.org/status/404');
+} catch (error) {
+	console.error(error.message);
```
Comment (szmarczak) [inline documentation/quick-start.md:130]: A better example would be: `error.response.statusCode`.

File: documentation/quick-start.md [hunk 19]
```diff
@@ -0,0 +1,236 @@
...
+		],
+	},
+};
+
+const client = got.extend(options);
+
+export default client;
+```
+
+*Note that handlers are given as arrays*, thus multiple handlers can be given. See documentation for other possible [hooks](9-hooks.md#hooks-api).
```
Comment (szmarczak) [inline documentation/quick-start.md:231]:
```suggestion
*Note that hooks are given as arrays*, thus multiple hooks can be given. See documentation for other possible [hooks](9-hooks.md#hooks-api).
```

Not to confuse with actual handlers: https://github.com/sindresorhus/got/blob/main/documentation/10-instances.md#gotextendoptions-instances

File: documentation/quick-start.md [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+
+### Going further
+
+There is a lot more to discover in the [documentation](../readme.md#documentation) and [tips](tips.md#tips).
+Among others, `Got` can handle [cookies](tips.md#cookies), [pagination](4-pagination.md#pagination-api), [cache](cache.md#cache). Read the documentation before implementing something that is already done by `Got` :innocent:.
```
Comment (szmarczak) [inline documentation/quick-start.md:236]:
```suggestion
Among others, `Got` can handle [cookies](tips.md#cookies), [pagination](4-pagination.md#pagination-api), [cache](cache.md#cache). Please read the documentation before implementing something that is already done by `Got` :innocent:.
```

File: documentation/quick-start.md (hunk 17 above)
Comment (ydarma) [reply documentation/quick-start.md:81]: You're right it is cleaner. I'll add a finally block to the `try { } catch { }`

File: documentation/quick-start.md (hunk 18 above)
Comment (ydarma) [reply documentation/quick-start.md:130]: done

File: documentation/quick-start.md (hunk 19 above)
Comment (ydarma) [reply documentation/quick-start.md:231]: Yes this is clear.

File: documentation/quick-start.md (hunk 20 above)
Comment (ydarma) [reply documentation/quick-start.md:236]: Ok, more pleasant for the reader.

File: documentation/quick-start.md (hunk 17 above)
Comment (szmarczak) [reply documentation/quick-start.md:81]: Do we need to use `fs.open` at all, why can't we just `fs.createWriteStream`?

Comment (ydarma) [reply documentation/quick-start.md:81]: The idea was to uniformly use the new promise API, as for streams. Before commit 6340f64ee92d32b9499cd721e25e9ab8ebc1d990 we used `fs.createWriteStream`. I don't mind, what do you think ? Less code is better ?

Comment (szmarczak) [reply documentation/quick-start.md:81]: > The idea was to uniformly use the new promise API, as for streams.

Let's not blindly use the APIs. What do you need the file descriptor for? If it's not needed then don't use it :)

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**